controllers/progressivesync_controller.go

```diff
@@ -267,6 +291,35 @@ func (r *ProgressiveSyncReconciler) getClustersFromSelector(ctx context.Context,
 	return secrets, nil
 }
 
+// calculateHashValue returns a hash value string for the ApplicationSet spec
+func (r *ProgressiveSyncReconciler) calculateHashedSpec(ctx context.Context, ps *syncv1alpha1.ProgressiveSync) (string, error) {
+	key := client.ObjectKeyFromObject(ps)
+	latest := syncv1alpha1.ProgressiveSync{}
+	if err := r.Client.Get(ctx, key, &latest); err != nil {
+		r.Log.Error(err, "Failed to retrieve the ProgressiveSync object while calculating the hash value")
+		return "", err
+	}
+
+	appSet := applicationset.ApplicationSet{}
+	if err := r.Client.Get(ctx, client.ObjectKey{Namespace: r.ArgoNamespace, Name: latest.Spec.SourceRef.Name}, &appSet); err != nil {
+		r.Log.Error(err, "Failed to retrieve the ApplicationSet object")
+		return "", err
+	}
+
+	appSetSpec := appSet.Spec
+	appSetSpecInBytes, err := json.Marshal(appSetSpec)
```

Comment:
```suggestion
	appSetSpecInBytes, err := json.Marshal(appSet.Spec)
```
It seems we only use `appSetSpec` so maybe we can just decode it?